roles/k3s_prep/tasks/main.yml

```
@@ -0,0 +1,62 @@
+---
+- name: Initialize /etc/rancher/k3s directory
+  ansible.builtin.file:
+    path: "/etc/rancher/k3s"
+    state: directory
+    mode: '0755'
+    owner: root
+    group: root
+
+- name: Make images directory
+  ansible.builtin.file:
+    path: "{{ k3s_server_location }}/agent/images/"
+    state: directory
+    mode: '0755'
+    owner: root
+    group: root
+
+- name: Create manifest directory
+  ansible.builtin.file:
+    path: "{{ k3s_server_location }}/server/manifests"
+    state: directory
+    mode: '0644'
+    owner: root
+    group: root
+
+- name: Create manifests directory for temp configuration
+  ansible.builtin.file:
+    path: "/tmp/k3s"
+    state: directory
+    owner: "{{ ansible_user_id }}"
+    mode: '0755'
+
+- name: Deploy k3s ARM
+  ansible.builtin.include_tasks: arm.yml
+  when:
+    - ( ansible_facts.architecture is search("arm") and
+      ansible_facts.userspace_bits == "64" ) or
+      ansible_facts.architecture is search("aarch64")
+
+- name: Deploy k3s x86_64
+  ansible.builtin.include_tasks: x86_64.yml
+  when: ansible_facts.architecture == "x86_64"
+
+# TODO: Move to sysctl
+- name: Enable IPv6 router advertisements
+  ansible.posix.sysctl:
+    name: net.ipv6.conf.all.accept_ra
+    value: "2"
+    state: present
+    reload: true
+  tags: sysctl
+
+# TODO: Unsure if this is needed at all
+- name: Add /usr/local/bin to sudo secure_path
+  ansible.builtin.lineinfile:
+    line: 'Defaults    secure_path = {{ secure_path[ansible_os_family] }}'
+    regexp: "Defaults(\\s)*secure_path(\\s)*="
+    state: present
+    insertafter: EOF
+    path: /etc/sudoers
+    validate: 'visudo -cf %s'
+  when: ansible_os_family in [ "Suse" ]
```

Use `ansible_distribution_file_variety`.

```suggestion
  when: ansible_distribution_file_variety == "SUSE"
```

`ansible_os_family` will change if the SUSE product differs. For example, SUSE's HPC product shows up as:
```json
        "ansible_os_family": "SLE_HPC",
```

---

Changed.